### PR TITLE
add sanity test for docker build and run

### PR DESCRIPTION
**Pull request by rbrugaro**
Add sanity test for docker build and run
@IlyasMoutawwakil this is in response to your request in closed PR #821

### CODE REVIEW

**Comment by IlyasMoutawwakil** [comment]
I think It's better to automate the sanity test by doing this in a GitHub workflow, that's triggered when the dockerfile changes.

**Comment by rbrugaro** [comment]
Thx @IlyasMoutawwakil, understood! I moved the build and run check to github workflow

File: .github/workflows/dockerfile_sanity.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Build and Test Docker Image
+
+on:
+  push:
+    paths:
+      - 'Dockerfile.intel'
+    branches:
+      - main
```
Comment (IlyasMoutawwakil) [inline .github/workflows/dockerfile_sanity.yml:8]:
```suggestion
on:
  push:
    branches:
      - main
    paths:
      - 'Dockerfile.intel'
 
  pull_request:
    branches:
      - main
    paths:
      - 'Dockerfile.intel'
```

File: .github/workflows/dockerfile_sanity.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+name: Build and Test Docker Image
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'Dockerfile.intel'
```
Comment (IlyasMoutawwakil) [inline .github/workflows/dockerfile_sanity.yml:8]: docker/Dockerfile.intel

Comment (rbrugaro) [reply .github/workflows/dockerfile_sanity.yml]: Thx! I used workflow_dispatch to manually trigger and test the workflow and forgot to update path for the auto triggering with file changes. Added!

**Pull request merged**